bundles/org.openhab.binding.evcc/src/main/java/org/openhab/binding/evcc/internal/api/EvccAPI.java

```diff
@@ -144,8 +148,12 @@ public String setVehicleLimitSoC(String vehicleName, int limitSoC) throws EvccAp
     }
 
     public String setVehiclePlan(String vehicleName, int planSoC, ZonedDateTime planTime) throws EvccApiException {
-        return httpRequest(this.host + EVCC_REST_API + "vehicles/" + vehicleName + "/plan/soc/" + planSoC + "/"
-                + planTime.toLocalDateTime().format(DateTimeFormatter.ISO_LOCAL_DATE_TIME) + "Z", "POST");
+        ZoneId zoneId = timeZoneProvider.getTimeZone();
+        ZonedDateTime adjustedTime = planTime.withZoneSameInstant(zoneId);
+        String formattedTime = adjustedTime.format(DateTimeFormatter.ISO_OFFSET_DATE_TIME);
```

Comment:
@Laith-Budairi - out of curiosity, does the API support UTC time, e.g. `2024-10-24T05:30:00.636Z`? In this case, the fix could be as simple as:
```java
String formattedTime = planTime.toInstant().toString();
```

And there would be no need to use `TimeZoneProvider`.